marc.py

```diff
@@ -0,0 +1,612 @@
+        else:
+            date_type = "n" # dates unknown
+            date_value = "    "
+        data += date_type + date_value
+        data += "    "
+        # TODO: Start tracking place of publication when available. Since we don't have
+        # this yet, assume everything was published in the US.
+        data += "xxu"
+        data += "                 "
+        language = "eng"
+        if edition.language and len(edition.language) == 3:
+            language = edition.language
+        data += language
+        data += "  "
+        record.add_field(
+            Field(tag="008", data=data))
+
+    @classmethod
+    def add_marc_organization_code(cls, record, marc_org):
+        record.add_field(
+            Field(tag="003", data=marc_org))
+
+    @classmethod
+    def add_isbn(cls, record, identifier):
+        # Add the ISBN if we have one.
+        isbn = None
+        if identifier.type == Identifier.ISBN:
+            isbn = identifier
+        if not isbn:
+            _db = Session.object_session(identifier)
+            for identifier_id in identifier.equivalent_identifier_ids()[identifier.id]:
+                equivalent = get_one(_db, Identifier, id=identifier_id)
+                if equivalent.type == Identifier.ISBN:
+                    isbn = equivalent
+                    break
+        if isbn:
+            record.add_field(
+                Field(
+                    tag="020",
+                    indicators=[" "," "],
+                    subfields=[
+                        "a", isbn.identifier,
+                    ]))
+
+    @classmethod
+    def add_title(cls, record, edition):
+        # Non-filing characters are used to indicate when the beginning of a title
+        # should not be used in sorting. This code tries to identify them by comparing
+        # the title and the sort_title.
+        non_filing_characters = 0
+        if edition.title != edition.sort_title and ("," in edition.sort_title):
+            sort_title = edition.sort_title
+            last_comma = len(sort_title) - sort_title[::-1].index(",")
+            stemmed = edition.sort_title[:last_comma-1]
+            non_filing_characters = edition.title.index(stemmed)
+        # MARC only supports up to 9 non-filing characters, but if we got more
+        # something is probably wrong anyway.
+        if non_filing_characters > 9:
+            non_filing_characters = 0
+
+        record.add_field(
+            Field(
+                tag="245",
+                indicators=["0", non_filing_characters],
+                subfields=[
+                    "a", unicode(edition.title) or "",
+                    "b", unicode(edition.subtitle) or "",
+                    "c", unicode(edition.author) or "",
+                ]))
+
+    @classmethod
+    def add_contributors(cls, record, edition):
+        """Create contributor fields for this edition.
+
+        TODO: Use canonical names from LoC.
```

Comment:
Are these the MARC relators? If so you can use `Contributor.MARC_ROLE_CODES`.

Reply:
This is referring to the contributor's names, not the roles, I'll update the comment.